File: ios/TrustlyViewManager.m
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(TrustlyViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(establish, NSDictionary)
RCT_EXPORT_VIEW_PROPERTY(hybrid, NSDictionary)
RCT_EXPORT_VIEW_PROPERTY(lazyOpenLightbox, BOOL)
RCT_EXPORT_VIEW_PROPERTY(selectBankWidget, NSDictionary)

RCT_EXPORT_VIEW_PROPERTY(onCancel, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onNotification, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onReturn, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onBankSelect, RCTDirectEventBlock)

@end

File: ios/TrustlyViewManager.swift
@objc(TrustlyViewManager)
class TrustlyViewManager: RCTViewManager {

  override func view() -> (TrustlyView) {
    return TrustlyView()
  }

  @objc override static func requiresMainQueueSetup() -> Bool {
    return false
  }
}

class TrustlyView : UIView {
    var lazy: Bool = false;
    
    @objc var onCancel: RCTDirectEventBlock?;
    @objc var onNotification: RCTDirectEventBlock?;
    @objc var onReturn: RCTDirectEventBlock?;
    @objc var onBankSelect: RCTDirectEventBlock?;
    
    @objc var establish: Dictionary<String, Any> = [:] {
      didSet {
          self.renderLightbox(data: establish);
      }
    }
    
    @objc var hybrid: Dictionary<String, Any> = [:] {
      didSet {
          self.renderHybrid(data: hybrid);
      }
    }
    
    @objc var lazyOpenLightbox: Bool = false {
      didSet {
        self.lazy = lazyOpenLightbox
      }
    }
    
    @objc var selectBankWidget: Dictionary<String, Any> = [:] {
      didSet {
          self.renderWidget(data: selectBankWidget);
      }
    }
    
    func renderHybrid(data: Dictionary<String, Any>) {
        print(data)
    }
    
    func renderLightbox(data: Dictionary<String, Any>) {
        print(data)
    }
    
    func renderWidget(data: Dictionary<String, Any>) {
        print(data)
    }
}
